Arc-i-Tech/Arc-i-Tech.github.io
# Create Reset Password API

## Create Reset Password API

## Description
This API allows a user to request a password reset using their username. The process includes validating the user's credentials, authenticating the user, and providing a response containing a password reset URL.

add the dependencies 
```
<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-security</artifactId>
</dependency>

<dependency>
    <groupId>org.springframework.boot</groupId>
    <artifactId>spring-boot-starter-mail</artifactId>
</dependency>

<dependency>
	<groupId>org.springframework.boot</groupId>
	<artifactId>spring-boot-devtools</artifactId>
</dependency>

```


## Type of Change
- [x] Project Configuration
- [ ] Repository Configuration
- [ ] Documentation update - Code guide/ docs (_like, javadoc or api_)


## How Has This Been Tested?/ Test Plan
  password reset test for manually and add the authentication concepts 
  after the authentication code and rest password API task merged then check the debugging code is worked 

 ## Checklist:

- [x] My code follows the project’s [coding guidelines](./../../docs/guidelines/coding-guides/). 
- [x] I have commented on my code where necessary.
- [x] I have added tests that verify the functionality of the code (if applicable).

## Screenshots / GIFs / Media (if applicable)
- NA

## How Has This Been Tested?/ Test Plan
- Send HTTP request  `username` and ` new password` through any Rest API client such as Postman.
- Check the data present in the object inside the use logger.
- Get HTTP response with login status, failure or success. 


## How to use/ enable
Send Password Reset Request

Use POST to call the URL:
http://localhost:8000/forgot_password/your_email_@gmail.com/your_username (use already registered username )
This request is made using Postman with the necessary parameters.
Access the Reset Link
After sending the request, check your email inbox for a password reset link.
Click the link in the email to open the reset URL.
Copy and Use the Reset URL in Postman
Copy the reset URL from the email and paste it into Postman.
Use GET to call the URL with the required parameters, such as token and new password:
new password add the reset_password  URL inside  like as 
http://localhost:8000/reset_password?token=XAi1WfjvKlq8OndKRqoHKGv5x8KoFd48UR02q0tJpl1Eh&password=add_your_new_password
Verify Success Message
After making the request, a message should appear:
"Password has been reset successfully."
Password Reset Confirmation

The password is now successfully updated.

## Additional Notes
In this code additional use of  updated login web handler #180  task code for reset password purpose  

Enable App Passwords for your Gmail account and use the generated app password instead of your actual Gmail password.
Steps to enable App Passwords:
Go to Google Account Settings.
Navigate to Security.
Under Signing in to Google, enable 2-Step Verification.
After enabling 2-Step Verification, you will see an App Passwords option.
 Generate one for your application.

## Closing issues
Closes #170
